File: Php/BMI.php
<?php
// Function to calculate BMI and determine category
function calculateBMI($weight, $height) {
    if ($height <= 0 || $weight <= 0) {
        return ["error" => "Invalid input values. Height and weight must be greater than zero."];
    }
    
    $bmi = $weight / ($height * $height);
    $category = "";

    // Determine BMI category
    if ($bmi < 18.5) {
        $category = "Underweight";
    } elseif ($bmi >= 18.5 && $bmi < 24.9) {
        $category = "Normal weight";
    } elseif ($bmi >= 25 && $bmi < 29.9) {
        $category = "Overweight";
    } else {
        $category = "Obese";
    }

    return ["bmi" => round($bmi, 2), "category" => $category];
}

// Prompt user for weight
echo "Enter your weight (kg): ";
$weight = trim(fgets(STDIN)); // Read input from user
$weight = floatval($weight); // Convert input to float

// Prompt user for height
echo "Enter your height (m): ";
$height = trim(fgets(STDIN)); // Read input from user
$height = floatval($height); // Convert input to float

// Calculate BMI
$result = calculateBMI($weight, $height);

// Display result
if (isset($result["error"])) {
    echo "Error: " . $result["error"] . "\n";
} else {
    echo "Your BMI is: " . $result["bmi"] . "\n";
    echo "Category: " . $result["category"] . "\n";
}
?>

File: Go/BMI.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// Function to calculate BMI and return the category
func calculateBMI(weight, height float64) (float64, string) {
	if height <= 0 || weight <= 0 {
		return 0, "Invalid input: Height and weight must be greater than zero."
	}

	bmi := weight / (height * height)
	var category string

	// Determine BMI category
	if bmi < 18.5 {
		category = "Underweight"
	} else if bmi >= 18.5 && bmi < 24.9 {
		category = "Normal weight"
	} else if bmi >= 25 && bmi < 29.9 {
		category = "Overweight"
	} else {
		category = "Obese"
	}

	return bmi, category
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	// Prompt user for weight
	fmt.Print("Enter your weight (kg): ")
	weightInput, _ := reader.ReadString('\n')
	weightInput = strings.TrimSpace(weightInput)
	weight, err := strconv.ParseFloat(weightInput, 64)
	if err != nil {
		fmt.Println("Invalid weight input. Please enter a number.")
		return
	}

	// Prompt user for height
	fmt.Print("Enter your height (m): ")
	heightInput, _ := reader.ReadString('\n')
	heightInput = strings.TrimSpace(heightInput)
	height, err := strconv.ParseFloat(heightInput, 64)
	if err != nil {
		fmt.Println("Invalid height input. Please enter a number.")
		return
	}

	// Calculate BMI
	bmi, category := calculateBMI(weight, height)

	// Display result
	if bmi == 0 {
		fmt.Println(category)
	} else {
		fmt.Printf("Your BMI is: %.2f\n", bmi)
		fmt.Println("Category:", category)
	}
}
